src/amalloy/ring_buffer.clj

```diff
@@ -47,7 +55,27 @@
       (RingBuffer. start (inc len) (assoc buf (rem (+ start len) (count buf)) x) meta)))
   (seq [this]
     (seq (for [i (range len)]
-           (nth buf (rem (+ start i) (count buf)))))))
+           (nth buf (rem (+ start i) (count buf))))))
+
+  Collection
+  (iterator [this]
+    (SeqIterator. (.seq this)))
+  (contains [this e]
+    (boolean (some #(= e %) (.seq this))))
+  (containsAll [this elts]
+    (every? #(.contains this %) elts))
+  (size [this]
+    (.count this))
+  (isEmpty [this]
+    (empty? this))
+  (toArray [this dest]
+    (reduce (fn [idx item]
+              (aset dest idx item)
+              (inc idx))
+            0, (.seq this))
```

Comment:
Don't call `.seq` here, just pass `this` and let `reduce` decide whether to seq or do something more efficient instead.